src/Analyzers/Core/Analyzers/SimplifyInterpolation/Helpers.cs

```diff
@@ -90,7 +90,9 @@ private static IOperation Unwrap(IOperation expression)
                     return;
                 }
 
-                if (invocation.Arguments.Length == 0)
+                if (invocation.Arguments.Length == 0 &&
+                    (invocation.TargetMethod.ContainingType.SpecialType == SpecialType.System_Object ||
+                    invocation.TargetMethod.OverriddenMethod?.ContainingType.SpecialType == SpecialType.System_Object))
```

Comment:
this won't be sufficient for the following:

```
class A
{
    public override string ToString() { }
}

class B : A
{
    public override string ToString() { }
}
```

Reply:
this looks like a situation where you want to check the "least overridden method" of the target method. Is there a helper for this available in this context?

Reply:
i'm aware of no such helper.  primarily because its' really trivial to just walk up.

Reply:
I couldn't find a helper either; this is the first instance in the namespace where `OverridenMethod` is used.